serpentTools/objects/containers.py

```diff
@@ -0,0 +1,144 @@
+
+
+class HomogUniv(SupportingObject):
+    """
+    Class for storing homogenized universe specifications and retrieving them
+
+     Parameters
+    ----------
+    -container: str
+        name of the parser
+    -name: str
+        name of the universe
+    -bu: float
+        burnup value
+    -step: float
+        temporal step
+    -day: float
+        depletion day
+
+    Attributes
+    ----------
+    -name: str
+        name of the universe
+    -bu: float
+        burnup value
+    -step: float
+        temporal step
+    -day: float
+        depletion day
+
+    """
+
+    def __init__(self, container, name, bu, step, day):
+        SupportingObject.__init__(self, container)
+        self.name = name
+        self.bu = bu
+        self.step = step
+        self.day = day
+        # Dictionaries:
+        self.b1Exp = {}
+        self.infExp = {}
+        self.b1Unc = {}
+        self.infUnc = {}
+        self.metaData = {}
+
+    def addData(self, variableName, variableValue, uncertainty=False):
+        """
+        Sets the value of the variable and, optionally, the associate s.d.
+
+        Parameters
+        ----------
+        variableName: str
+            Variable Name
+        variableValue: str/tuple/list
+            Variable Value
+        uncertainty:   bool
+            Boolean Variable- set to True in order to retrieve the
+            uncertainty associated to the expected values
+
+        Notes:
+        ---------
+            Raises a warning if the value of the variable is overwritten
+
+        """
+
+        # 1. Check the input type
+        variableName = SupportingObject._convertVariableName(variableName)
+        if not isinstance(uncertainty, bool):
+            raise messages.error('The variable uncertainty has type %s.\n ...'
+                                 'It should be boolean.', type(uncertainty))
+        # 2. Pointer to the proper dictionary
+        setter = self._lookup(variableName, uncertainty)
+        # 3. Check if variable is already present. Then set the variable.
+        if variableName in setter:
+            messages.warning(
+                "The variable {} will be overwritten".format(variableName))
+        setter[variableName] = variableValue
+
+    def get(self, variableName, uncertainty=False):
+        """
+        Gets the value of the variable VariableName from the dictionaries
+
+        Parameters
+        ----------
+        variableName: str
+            Variable Name
+        uncertainty:   bool
+            Boolean Variable- set to True in order to retrieve the
+            uncertainty associated to the expected values
+
+        Returns
+        -------
+        x: str/tuple/list
+            Variable Value
+        dx: str/tuple/list
+            Associated uncertainty
+
+        """
+        # 1. Check the input values
+        if not isinstance(uncertainty, bool):
+            raise messages.error('The variable uncertainty has type %s.\n ...'
```

Comment:
Have this raise a [`TypeError`](https://docs.python.org/2/library/exceptions.html#exceptions.TypeError) with the same message instead. Same with the call in the `addData` method

Reply:
Done